src/LieTheory/WeylGroup.jl

```diff
@@ -284,7 +284,12 @@ end
 @doc raw"""
     torsion_subgroup(W::WeylGroup) -> WeylGroup, Map
 
-Return the torsion subgroup of `W`.
+Return the torsion subgroup of `W`, i.e. the subgroup of all elements of `W` with finite order.  
+
+For a finite Weyl group, every element has finite order, so the torsion subgroup
+is the entire group `W`.
+
+If `W` is infinite, an `NotImplementedError` exception will be thrown.
```

Comment:
Ok, so with this definition, an infinite Weyl group will not have a torsion subgroup as the set of all elements of finite order do not form a subgroup. So maybe you can throw some more appropriate error (similar to the GAPGroup case) here?